sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -271,7 +279,20 @@ bool Scheduler::removeMemoryObject(detail::SYCLMemObjI *MemObj,
     // No operations were performed on the mem object
     return true;
 
-  {
+#ifdef _WIN32
+  // If we are shutting down on Windows it may not be
+  // safe to wait on host threads, as the OS may
+  // abandon them. But no worries, the memory WILL be reclaimed.
+  const bool allowWait =
+      MemObj->hasUserDataPtr() || GlobalHandler::instance().isOkToDefer();
```

Comment:
is there any sense in hasUserDataPtr check when the program finishes execution?

Reply:
No, but we have tests that, for whatever reason, don't use .wait() but they clear the leak check because they DO clean up before the UR adapters are released.  
Also I'm trying to limit this behavior change to only when it is needed, ie Windows, after main(), dealing with host threads.